src/blueapi/worker/reworker.py

```diff
@@ -68,6 +68,10 @@ def __init__(
     def submit_task(self, name: str, task: Task) -> None:
         active_task = ActiveTask(name, task)
         LOGGER.info(f"Submitting: {active_task}")
+        if self._task_queue.qsize() != 0:
+            LOGGER.error("Cannot submit task while another is running")
+            raise Exception("Cannot submit task while another is running")
```

Comment:
Would be good if we could use a more specific type of exception. It's a shame there's no `BusyError` or similar. Feel free to make one up if you can't find something close enough.